storage/cloud-client/notification_polling.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to poll for GCS notifications from a
+Cloud Pub/Sub subscription, parse the incoming message, and acknowledge the
+successful processing of the message.
+
+This application will work with any subscription configured for pull rather
+than push notifications. If you do not already have notifications configured,
+you may consult the docs at
+https://cloud.google.com/storage/docs/reporting-changes or follow the steps
+below:
+
+1. [Activate the Google Cloud Pub/Sub API], if you have not already done so.
+2. Create a Google Cloud Storage bucket:
+
+    ```
+    $ gsutil mb gs://testbucket
+    ```
+
+3. Create a Cloud Pub/Sub topic and publish bucket notifications there:
+
+   ```
+   $ gsutil notification create -f json -t testtopic gs://testbucket
+   ```
+
+4. Create a subscription for your new topic:
+
+   ```
+   $ gcloud beta pubsub subscriptions create testsubscription --topic=testtopic
+   ```
+
+5. Run this program:
+
+   ```
+   $ python notification_polling testsubscription`
+   ```
+
+6. While the program is running, upload and delete some files in the testbucket
+   bucket (you could use the console or gsutil) and watch as changes scroll by
+   in the app.
+"""
+
+import argparse
+import json
+import sys
+
+from cachetools import LRUCache
+from google.cloud import pubsub
+
+
+ACTION_STRINGS = {
+    'OBJECT_ARCHIVE': 'Object archived',
+    'OBJECT_DELETE': 'Object deleted',
+    'OBJECT_FINALIZE': 'Object created',
+    'OBJECT_METADATA_UPDATE': 'Object metadata updated'
+}
+
+
+DUPLICATE_EVENT_CACHE = LRUCache(10000)
+DUPLICATION_KEY = '{event_type}|{resource}|{metageneration}'
+
+
+def Summarize(message):
+    # [START parse_message]
+    data = message.data
+    attributes = message.attributes
+
+    # The kind of event that just happened. Example: OBJECT_FINALIZE
+    event_type = attributes['eventType']
+    event_description = ACTION_STRINGS.get(
```

Comment:
Every indirection you put in a sample is a place where users can get lost or confused. Two questions:

1. Is the list of event types in the documentation somewhere? If so, can you just print out the `eventType` and add a comment with a link to that documentation?
2. If the event types are not in the documentation somewhere, can you move the action strings stuff into this block?